cpp/src/Ice/Communicator.cpp

```diff
@@ -110,13 +110,18 @@ Ice::Communicator::identityToString(const Identity& ident) const
 }
 
 ObjectAdapterPtr
-Ice::Communicator::createObjectAdapter(const string& name)
+Ice::Communicator::createObjectAdapter(
+    const string& name,
+    const optional<SSL::ServerAuthenticationOptions>& serverAuthenticationOptions)
```

Comment:
Say the user doesn't specify a serverAuthenticationOptions. Per your PR description, we inject instead a config-generated serverAuthenticationOptions, presumably created using SSLEngine::createServerAuthenticationOptions.

I can't figure out where we perform this injection.

Reply:
See 

https://github.com/zeroc-ice/ice/blob/f84b577346b670be9c24960f24d5ad6e50d2c788/cpp/src/IceSSL/SSLAcceptorI.cpp#L69-L73

And

https://github.com/zeroc-ice/ice/blob/f84b577346b670be9c24960f24d5ad6e50d2c788/cpp/src/IceSSL/SSLConnectorI.cpp#L36-L41